### PR TITLE
feat!: convert to ESM and vitest

### LINKED ISSUE #749: Use pure ES modules
See background at https://github.com/netlify/team-dev/issues/36

Once https://github.com/netlify/zip-it-and-ship-it/issues/748 is done and released, we should use pure ES modules and make a major release.

This is more than just switching from CommonJS to `import`/`export`. See [this list](https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c) for other changes which might be involved. This should be broken in many PRs, as much as possible, to lower the risk. Also, non-breaking changes (such as adding file extensions in imports, or loading JSON files differently) should be done before the breaking changes (such as using `import`/`export` statements).

This should only be done after pure ES modules are supported in both Netlify CLI (https://github.com/netlify/cli/issues/3514) and Netlify Build (https://github.com/netlify/build/issues/3742).

### PR DESCRIPTION
**Pull request by danez**
🎉 Thanks for submitting a pull request! 🎉

#### Summary

Fixes #749 

This converts zisi to pure ESM. It also migrates from ava to vitest.

---

For us to review and ship your PR efficiently, please perform the following steps:

- [ ] Open a [bug/issue](https://github.com/netlify/zip-it-and-ship-it/issues/new/choose) before writing your code 🧑‍💻.
      This ensures we can discuss the changes and get feedback from everyone that should be involved. If you\`re fixing
      a typo or something that\`s on fire 🔥 (e.g. incident related), you can skip this step.
- [ ] Read the [contribution guidelines](../CONTRIBUTING.md) 📖. This ensures your code follows our style guide and
      passes our tests.
- [ ] Update or add tests (if any source code was changed or added) 🧪
- [ ] Update or add documentation (if features were changed or added) 📝
- [ ] Make sure the status checks below are successful ✅

**A picture of a cute animal (not mandatory, but encouraged)**

### CODE REVIEW

File: .github/workflows/release-please.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ jobs:
         with:
           token: ${{ steps.get-token.outputs.token }}
           release-type: node
-          package-name: '@netlify/@netlify/zip-it-and-ship-it'
+          package-name: '@netlify/zip-it-and-ship-it'
```
Comment (lukasholzer) [inline .github/workflows/release-please.yml:20]: how did this worked 😂

File: benchmarks/esbuild.js [hunk 2]
```diff
@@ -19,4 +20,9 @@ const runBenchmarks = async function () {
   console.log(`${largeDepsEsbuild}ms`)
 }
 
-runBenchmarks()
+try {
+  await runBenchmarks()
```
Comment (lukasholzer) [inline benchmarks/esbuild.js:24]: Just as information you could now use the `vitest bench` benchmarking functionality. Nothing to change here but just as a FYI

File: src/archive.ts [hunk 3]
```diff
@@ -8,6 +8,7 @@ import endOfStream from 'end-of-stream'
 
 export { Archiver as ZipArchive } from 'archiver'
 
+// TODO make enum
```
Comment (lukasholzer) [inline src/archive.ts:11]: Still todo?

**Review by lukasholzer — APPROVED** [review]
❤️ typescript 💯

File: src/runtimes/node/bundlers/nft/index.ts [hunk 4]
```diff
@@ -13,6 +13,11 @@ import type { GetSrcFilesFunction, BundleFunction } from '../types.js'
 
 import { processESM } from './es_modules.js'
 
+// @ts-expect-error vitest with esbuild already resolves the default export, whereas Node.js does not
+const resolveDependency: typeof nftResolveDependency.default = nftResolveDependency.default
+  ? nftResolveDependency.default
+  : nftResolveDependency
```
Comment (danez) [inline src/runtimes/node/bundlers/nft/index.ts:19]: Reported here: https://github.com/vitest-dev/vitest/issues/2219

File: tests/helpers/main.ts [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+
+  return result
+}
+
+// Import a file exporting a function.
+// Returns `default` exports as is.
+export const importFunctionFile = async function <T = any>(functionPath: string): Promise<T> {
+  // We use relative paths here, because vitest cannot handle absolute paths or urls :(
+  // eslint-disable-next-line import/no-dynamic-require
+  const result = await import(`/@fs/${functionPath}`)
```
Comment (danez) [inline tests/helpers/main.ts:146]: Reported here https://github.com/vitest-dev/vitest/issues/2218

File: .github/workflows/release-please.yml (hunk 1 above)
Comment (danez) [reply .github/workflows/release-please.yml:20]: No Idea

File: benchmarks/esbuild.js (hunk 2 above)
Comment (danez) [reply benchmarks/esbuild.js:24]: Yeah, I was wondering, but didn't want to add even more changes.
Not super happy with vitest yet, had to do some workarounds, so do not want to buy in yet fully :)

File: src/archive.ts (hunk 3 above)
Comment (danez) [reply src/archive.ts:11]: Yes, afterwards.

**Pull request merged**